crates/oxc_linter/src/rules/import/no_duplicates.rs

```diff
@@ -131,7 +133,11 @@ impl Rule for NoDuplicates {
                             _ => 0,
                         }
                     };
-                    import_entries_maps.entry(key).or_default().push(requested_module);
+
+                    if flags[key as usize] {
+                        flags[key as usize] = false;
```

Comment:
Could we convert `key` to be `u8` instead to avoid sign loss here?

```diff
-ImportImportName::Name(_) => i8::from(!self.prefer_inline),
+ImportImportName::Name(_) => u8::from(!self.prefer_inline),
```